AmplifyPlugins/DataStore/AWSDataStoreCategoryPluginIntegrationTests/DataStoreConsecutiveUpdatesTests.swift

```diff
@@ -0,0 +1,331 @@
+                return
+            }
+
+            guard let post = try? mutationEvent.decodeModel() as? Post,
+                  post.id == myPost.id else {
+                return
+            }
+
+            if mutationEvent.mutationType == GraphQLMutationType.create.rawValue {
+                XCTAssertEqual(post.title, "MyPost")
+                XCTAssertEqual(post.content, "This is my post.")
+                XCTAssertEqual(post.rating, 3)
+                saveSyncExpectation.fulfill()
+                return
+            }
+
+            if mutationEvent.mutationType == GraphQLMutationType.delete.rawValue {
+                XCTAssertEqual(post.title, "MyPost")
+                XCTAssertEqual(post.content, "This is my post.")
+                XCTAssertEqual(post.rating, 3)
+                deleteSyncExpectation.fulfill()
+                return
+            }
+        }
+
+        guard try HubListenerTestUtilities.waitForListener(with: hubListener, timeout: 5.0) else {
+            XCTFail("Listener not registered for hub")
+            return
+        }
+
+        // save the post and wait for it to sync
+        let immediateSaveExpectation = expectation(description: "Post is saved")
+        Amplify.DataStore.save(myPost) { result in
+            switch result {
+            case .success:
+                immediateSaveExpectation.fulfill()
+            case .failure(let error):
+                XCTFail("Error: \(error)")
+            }
+        }
+        wait(for: [immediateSaveExpectation], timeout: networkTimeout)
+        wait(for: [saveSyncExpectation], timeout: networkTimeout)
+
+        // delete the updated post
+        let immediateDeleteExpectation = expectation(description: "Post is immediately deleted")
+        Amplify.DataStore.delete(myPost) { result in
+            switch result {
+            case .success:
+                immediateDeleteExpectation.fulfill()
+            case .failure(let error):
+                XCTFail("Error: \(error)")
+            }
+        }
+        wait(for: [immediateDeleteExpectation], timeout: networkTimeout)
+        wait(for: [deleteSyncExpectation], timeout: networkTimeout)
+
+        // query the deleted post
+        let queryExpectation = expectation(description: "Post is not found")
+        Amplify.DataStore.query(Post.self, byId: myPost.id) { result in
+            switch result {
+            case .success(let post):
+                XCTAssertNil(post)
+                queryExpectation.fulfill()
+            case .failure(let error):
+                XCTFail("Error: \(error)")
+            }
+        }
+        wait(for: [queryExpectation], timeout: networkTimeout)
+
+    }
+
+    /// - Given: API has been setup with `Post` model registered
+    /// - When: A Post is saved with sync complete, updated and deleted immediately
+    /// - Then: The Post should not be returned when queried for
+    func testSaveUpdateAndImmediateDelete() throws {
```

Comment:
`testSaveThenUpdateAndImmediatelyDelete`

Reply:
nit: reorder this test case as the 3rd one
1. testSaveAndImmediatelyUpdate
2. testSaveAndImmediatelyDelete
3. testSaveThenUpdateAndImmediatelyDelete